### ISSUE TITLE
Better lane/segment/node debugger

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#844]
Our current approach to debugging lanes/segments/nodes is just showing _way_ too much info at once:

![spam](https://user-images.githubusercontent.com/1386719/79397303-b0a92e00-7f75-11ea-98e5-4f85f0fdbf5c.png)

When you zoom in, the labels start moving towards the actual nodes/segments/lanes - ie. the zoomed out view is misleading becacuse labels are in wrong places. So this tool is just a PITA to work with.

A better approach would be to use lane highlighting (same as we do in various TMPE tools where we draw bezier shape over lanes/segments/nodes) but for all networks on screen.

What needs to be shown:

* Nodes, segments and lanes of any network
* (De)Spawn points for vehicles / cims?
* Maybe some way to filter to specific network types (vehicle, pedestrian, etc)?

Color coding depending on type of network:

* Paths in buildings:
    * _See Touch This Tool mod by klyte45 for how to detect; also Move It mod can select them while Alt is pressed._
    * Orange = vehicle paths
    * Purple - pedestrian paths _(they are shown purple in asset editor)_
* Paths outside buildings:
    * Green - vehicle paths
    * Blue - pedestrian paths
* Anything else:
    * Red

Interaction:

* Click a highlighted lane/segment/node = show world info panel.
    * If Mod Tools mod is active, it automatically attaches the buttons to scene explorer which is useful
* Some way to dump data about the selected thing to log file
    * Basically a full parameter dump of the thing and also any other things it contains (eg. a segment would also list its lanes)

I think this would make debugging much easier, and also leave us with a load of code that could be used for other debug tools at later date.

**Comment by kianzarrin** [comment]
I always just use modtools.

**Comment by aubergine10** [comment]
mod tools is fine when you know what you're looking for and can select it

but for general debugging it would be helpful to just have a way to quickly highlight all the net stuff and click on it to get its worldinfo panel (which in turn has the mod tools buttons)

modtools can sometimes be cumbersome so some 'dump this thing to log' would be useful too

**Comment by kianzarrin** [comment]
![image](https://user-images.githubusercontent.com/26344691/86453557-adc50a80-bd26-11ea-894d-04383b88ab66.png)
@aubergine10 
I am working on a mod to help with getting information about networks. more easily than mod tools.
i call it network detective
Features:
hover over or select nodes/segment to get details.
Hover over each of the elements in the display panel to get details about that particular element

objective: help with writing code (not to design assets).

Focus: present important traffic related information of networks.

**Comment by aubergine10** [comment]
This is awesome! Can it be extended to ped paths, quays and other networks? I'm particularly interested in quays because currently I have no idea which quays include embedded ped paths (feature added to game in Sunset Harbor).

**Comment by aubergine10** [comment]
Also, in the dialog that lists the lanes, would it be possible to include lane type (eg. vehicle) so it's a bit easier to spot at a glance which lanes can hold traffic, etc?

**Comment by aubergine10** [comment]
Closing this as Network Detective does all this stuff now (TMPE overlays can be used to find ids of unsual lanes/segments/etc and then use search feature in ND mod to get details of them).

**Comment by kianzarrin** [comment]
> @aubergine10  Network Detective does all this stuff now 

ND doesn't monitor traffic value, ... , We need a TMPE version of Network detective ;)

Should we open this again?

**Comment by aubergine10** [comment]
The existing overlays are useful enough for traffi value, I don't think we need that in ND.
